### ISSUE TITLE
pane content and zellij ui not redrawing correctly

### ISSUE THREAD

**Issue by LU15W1R7H** [zellij-org/zellij#1023]
When using zellij I oftentimes experience graphical bugs, where zellij fails to redraw the contents of a pane or the zellij menubar.
I believe this problem started after the upgrade to 0.24 because earlier I didn't notice anything like it.

I see it most often when using [nnn](https://github.com/jarun/nnn). When moving the cursor, zellij failes to repaint and filenames disappear or the cursor is at multiple places at once.

Screenshot after opening a text file using nvim in nnn and closing it again, nnn doesn't show the filenames anymore. But the menubar of nnn is there. The menubar of zellij is also distorted.
The attached zellij log files are generated by the session in this screenshot.
![2022-01-24_164029](https://user-images.githubusercontent.com/37505890/150816852-a7d46998-d2ac-4e30-adb1-36bbdd35cf10.png)

Screenshot of another example where the zellij menubar isn't redrawn properly.
![2022-01-24_163824](https://user-images.githubusercontent.com/37505890/150817292-4473c78c-a75d-4070-bc07-9e154be77fff.png)

```
> uname -av
Linux lwirth-tarch 5.16.2-arch1-1 #1 SMP PREEMPT Thu, 20 Jan 2022 16:18:29 +0000 x86_64 GNU/Linux
> kitty --version
kitty 0.24.1 created by Kovid Goyal
> zellij --version
zellij 0.24.0
> nnn -V
4.4
```

[/tmp/zellij-1000/zellij-log/zellij.log](https://github.com/zellij-org/zellij/files/7926772/zellij.log)
[/tmp/zellij-1000/zellij-log/zellij-10.log](https://github.com/zellij-org/zellij/files/7926773/zellij-10.log)

**Comment by imsnif** [comment]
Hum - super weird! How did you install Zellij?

**Comment by LU15W1R7H** [comment]
@imsnif Using `pacman -S zellij`

**Comment by LU15W1R7H** [comment]
As zellij is quite unusable for me this way, I was wondering if there is a way to force zellij to redraw more often.

**Comment by imsnif** [comment]
You can resize the terminal window...

Does this happen with other terminal emulators? eg. Alacritty/gnome-terminal?

Can you show the output of `zellij setup --check` here?

**Comment by malthesr** [comment]
I get something similar in kitty since upgrading to 0.24. I think it's the same issue, so I'm adding my info here instead of creating a new issue. 

If I open zellij, create a new pane, and decrease its size, my prompt location gets updated, but the old location is not removed (along with the pane border next to the prompt). After decreasing a few times, it looks like so:

![zellij](https://user-images.githubusercontent.com/25851939/151815845-45aebbf2-17a7-4f7b-8626-4262bc0e4014.png)

I don't seem to get the same issue with horizontal splits.

Running `zellij --debug` and reproducing the above, the only non-empty log I get in `/tmp/zellij-log/` is the main `zellij.log` file, which doesn't reveal much, at least to me (lightly redacted):

```
INFO   |zellij_client [...] zellij-client-0.24.0/src/lib.rs:118]: Starting Zellij client! 
INFO   |zellij_server [...] zellij-server-0.24.0/src/lib.rs:186]: Starting Zellij server! 
INFO   |zellij_server::wasm_vm [...] zellij-server-0.24.0/src/wasm_vm.rs:86]: Wasm main thread starts 
INFO   |zellij_server::wasm_vm [...] zellij-server-0.24.0/src/wasm_vm.rs:226]: wasm main thread exits 
INFO   |zellij_client [...] zellij-client-0.24.0/src/lib.rs:342]: Bye from Zellij! 
```

I did not observe this in 0.23. I also cannot reproduce in alacritty.

I'm on Manjaro (xfce) 5.10.93-1. Additional info:

```shell
> zellij --version
zellij 0.24.0
> tput lines
52
> tput cols
213
> kitty --version
kitty 0.24.1 created by Kovid Goyal
```

**Comment by malthesr** [comment]
For what it's worth, behaviour changed slightly for me here after upgrading to 0.25. Now the prompt itself gets redrawn slightly better when changing pane size, but the top of the Zellij pane is off. Compare the screenshot below to the one above:

![zellij25](https://user-images.githubusercontent.com/25851939/155148884-da639974-d7d5-41ac-a866-0e47853f4748.png)

Just posting in case this helps to diagnose the issue.

**Comment by malthesr** [comment]
This seems to have been fixed for me in 0.26. Thank you.

**Comment by a-kenji** [comment]
That is great to hear, thank you for checking!
